packages/react-components/src/graph-visualizer/svg.ts

```diff
@@ -0,0 +1,185 @@
+import type { NodeSingular } from 'cytoscape';
+
+import {
+  NODE_HEXAGON_HEXAGON_POINTS,
+  NODE_DIAMOND_DEFAULT_SIZE,
+  NODE_HEXAGON_DEFAULT_SIZE,
+  NODE_ELLIPSE_DEFAULT_SIZE,
+  NODE_RECTANGLE_DEFAULT_SIZE,
+} from './constants';
+import type { NodeRenderData, NodeShape, NodeStyleProps } from './types';
+
+/**
+ * Returns a SVG string for the `shape`. Defaults to `ellipse`.
+ */
+export function createSvgString({
+  backgroundColor,
+  borderColor,
+  borderWidth,
+  shape,
+  size,
+}: Pick<NodeStyleProps, 'backgroundColor' | 'borderColor' | 'borderWidth'> & {
+  shape?: NodeShape;
+  size: number;
+}) {
+  switch (shape) {
+    case 'diamond':
```

Comment:
Is the cytoscape NodeShape type have enums that we can do the case matches on instead of strings? If they're using a union string type then this is fine as is.

Reply:
They are using union string type,